ui/pages/settings/contact-list-tab/contact-list-tab.component.js

```diff
@@ -1,16 +1,57 @@
 import React, { Component } from 'react';
+import { useDispatch } from 'react-redux';
 import PropTypes from 'prop-types';
-import classnames from 'classnames';
 import ContactList from '../../../components/app/contact-list';
-import {
-  CONTACT_ADD_ROUTE,
-  CONTACT_VIEW_ROUTE,
-} from '../../../helpers/constants/routes';
-import Button from '../../../components/ui/button';
+import { showModal } from '../../../store/actions';
+import { CONTACT_VIEW_ROUTE } from '../../../helpers/constants/routes';
 import EditContact from './edit-contact';
 import AddContact from './add-contact';
 import ViewContact from './view-contact';
 
+AddContactButton.propTypes = {
+  label: PropTypes.string,
+};
+
+RenderAddContactLink.propTypes = {
+  label: PropTypes.string,
+};
+
+function RenderAddContactLink(props) {
+  const { label } = props;
+  const dispatch = useDispatch();
+
+  return (
+    <div>
+      <button
+        className="address-book__link"
+        onClick={(e) => {
+          e.preventDefault();
+          dispatch(showModal({ name: 'ADD_NEW_CONTACT' }));
+        }}
+      >
+        + {label}
+      </button>
+    </div>
+  );
+}
+
+function AddContactButton(props) {
```

Comment:
[REQ] We should maintain only one component per file. As this is a separate react component, we would need to move it to its own file. Although I'm not sure much is gained by separating out this component from the JSX tree. Maybe just render the button inline instead of making it separate?